### PR TITLE
Added watsonmarlow323 device for i22 ppump.

### PR DESCRIPTION
**Pull request by barnettwilliam**
Added Watson Marlow 323 pump device for use on I22 - draft PR for concept review.

This device depends on an updated support module interface (support/watson-marlow) that was made more consistent while making this change - release TBC.

### Instructions to reviewer on how to test:
1.  TBC
2. Confirm that device is operable 

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)
- [ ] If changing the API for a pre-existing device, ensure that any beamlines using this device have updated their Bluesky plans accordingly
- [ ] Have the connection tests for the relevant beamline(s) been run via `dodal connect ${BEAMLINE}`

### CODE REVIEW

File: src/dodal/devices/watsonmarlow323.py
Comment (callumforrester) [inline src/dodal/devices/watsonmarlow323.py:18]: Nit: I would prefer a name that gives some suggestion as to what it does, maybe:

```suggestion
class WatsonMarlow323Pump(StandardReadable):
```

File: tests/devices/unit_tests/test_watsonmarlow323.py
Comment (callumforrester) [inline tests/devices/unit_tests/test_watsonmarlow323.py:36]: Should: There is a utility function in `ophyd-async` that does this, you can just import that

Example: https://github.com/DiamondLightSource/dodal/blob/main/tests/devices/unit_tests/test_undulator.py

File: src/dodal/devices/watsonmarlow323.py
Comment (callumforrester) [inline src/dodal/devices/watsonmarlow323.py:22]: I haven't read the manual, but is this PV really read-only?

File: src/dodal/devices/watsonmarlow323.py
Comment (callumforrester) [inline src/dodal/devices/watsonmarlow323.py:12]: Nit: More meaningful names would be good here, we need to retain the cryptic acronyms on the epics side, but the ophyd side can have long, descriptive names e.g.

```suggestion
class WatsonMarlow323Direction(str, Enum):
     CLOCKWISE = "CW"
     COUNTER_CLOCKWISE = "CCW"
```

**Review by callumforrester — CHANGES_REQUESTED** [review]
Looking good, thanks @barnettwilliam 

I've made a few comments, mostly minor stuff.

You also need to rebase against `main` and fix the linting. 
You can run `ruff format` and `ruff check --fix` to do that.
Additionally, if you use vscode, I recommend copy/pasting in [this settings directory](https://github.com/DiamondLightSource/python-copier-template/tree/main/.vscode) (it's gitignored) to make it auto format etc.

File: src/dodal/devices/watsonmarlow323.py
Comment (barnettwilliam) [reply src/dodal/devices/watsonmarlow323.py]: Renamed as suggested also updated filenames in 417aaa12897e1aad61c73db30d4f3d6e00974e29.

File: tests/devices/unit_tests/test_watsonmarlow323.py
Comment (barnettwilliam) [reply tests/devices/unit_tests/test_watsonmarlow323.py]: Updated to use the ophyd-async definition as suggested in 417aaa12897e1aad61c73db30d4f3d6e00974e29, `assert_reading()` is defined in `test_slits.py` the file I initially looked at when creating the test for this device.

File: src/dodal/devices/watsonmarlow323.py
Comment (barnettwilliam) [reply src/dodal/devices/watsonmarlow323.py]: Checking this its not read only so I've made it rw in 417aaa12897e1aad61c73db30d4f3d6e00974e29.

File: src/dodal/devices/watsonmarlow323.py
Comment (barnettwilliam) [reply src/dodal/devices/watsonmarlow323.py]: Updated names in 417aaa12897e1aad61c73db30d4f3d6e00974e29

**Comment by barnettwilliam** [comment]
Thanks @callumforrester I've incorporated your suggested changes, rebased, and linted. Also, made `enabled` a readable signal based on @DiamondJoseph's comments following chat on Wed. 

 I just need to test this with an actual device to confirm it communicates correctly after which I'll undraft and submit.

File: src/dodal/beamlines/i22.py
Comment (DiamondJoseph) [inline src/dodal/beamlines/i22.py:315]: Why is this device skipped for i22?

File: src/dodal/beamlines/i22.py
Comment (DiamondJoseph) [inline src/dodal/beamlines/i22.py:325]: Please also add to p38.py with `fake_with_ophyd_sim: bool = True`

File: src/dodal/devices/watsonmarlow323_pump.py
Comment (DiamondJoseph) [inline src/dodal/devices/watsonmarlow323_pump.py:29]: write_pv isn't required if they're both the same.

File: src/dodal/devices/watsonmarlow323_pump.py
Comment (DiamondJoseph) [inline src/dodal/devices/watsonmarlow323_pump.py:50]: Limits (and units, precision...) will propagate with the speed signal once [this change is merged](https://github.com/bluesky/ophyd-async/pull/137), these signals shouldn't be required: as is, it's not clear what they are the limits of, but when the change is merged they'll become metadata on the speed signal.

File: src/dodal/devices/watsonmarlow323_pump.py
Comment (DiamondJoseph) [inline src/dodal/devices/watsonmarlow323_pump.py:59]: Nit: This and the above signal creation can be replaced with the new ContextManager on StandardReadable:

```suggestion
        with self.add_children_as_readables():
          self.direction = epics_signal_rw(
            WatsonMarlow323PumpDirection,
            read_pv=prefix + "INFO:DIR",
            write_pv=prefix + "SET:DIR",
          )
          self.state = epics_signal_rw(
            WatsonMarlow323PumpState,
            read_pv=prefix + "INFO:RUN",
            write_pv=prefix + "SET:RUN",
          )
          self.speed = epics_signal_rw(
            float, read_pv=prefix + "INFO:SPD", write_pv=prefix + "SET:SPD"
          )
        with self.add_children_as_readables(ConfigSignal):
          self.enabled = epics_signal_rw(
            WatsonMarlow323PumpEnable,
            read_pv=prefix + "DISABLE")
```

Comment (barnettwilliam) [reply src/dodal/devices/watsonmarlow323_pump.py:59]: I did initially consider the using the the context manager as shown in the tutorial but it wasn't clear to me from the name of HintedSignal what it was and I couldn't find any documentation clarifying this. `read` and `config` made more sense and there were more existing examples using this format.

The next device I create I'll use the context manager.

Comment (callumforrester) [reply src/dodal/devices/watsonmarlow323_pump.py:59]: https://github.com/bluesky/ophyd-async/issues/349

File: src/dodal/devices/watsonmarlow323_pump.py
Comment (barnettwilliam) [reply src/dodal/devices/watsonmarlow323_pump.py:50]: Ok I'll remove these when bluesky/ophyd-async#137 is merged

File: src/dodal/beamlines/i22.py
Comment (DominicOram) [inline src/dodal/beamlines/i22.py:316]: Is this deliberately double p? If so what does the first p stand for?

Comment (Tom-Willemsen) [reply src/dodal/beamlines/i22.py:316]: Interjecting here, the double p matches what's on i22's synoptic (it feels like matching the synoptic name <-> bluesky name makes sense?).

The p is for peristaltic pump I think, as opposed to spump which is a syringe pump I believe.

Comment (DominicOram) [reply src/dodal/beamlines/i22.py:316]: Can we put this in the docstring? Otherwise people will (rightly) think it looks like a typo

Comment (callumforrester) [reply src/dodal/beamlines/i22.py:316]: Or could call the device `peristaltic_pump`

File: src/dodal/beamlines/i22.py
Comment (barnettwilliam) [reply src/dodal/beamlines/i22.py]: Fixed in 6d5ae3fb866bdad8de27966fc11605b6c41856cd

File: src/dodal/beamlines/i22.py
Comment (barnettwilliam) [reply src/dodal/beamlines/i22.py]: I didn't want interfere with anything, re-enabled in 6d5ae3fb866bdad8de27966fc11605b6c41856cd

File: src/dodal/devices/watsonmarlow323_pump.py
Comment (barnettwilliam) [reply src/dodal/devices/watsonmarlow323_pump.py]: Fixed in 6d5ae3fb866bdad8de27966fc11605b6c41856cd

File: src/dodal/beamlines/i22.py
Comment (barnettwilliam) [reply src/dodal/beamlines/i22.py:315]: Added docstring description in 6d5ae3fb866bdad8de27966fc11605b6c41856cd, I'll find out what the scientists would prefer as the device name.

File: src/dodal/beamlines/i22.py
Comment (barnettwilliam) [reply src/dodal/beamlines/i22.py]: Oops, thanks corrected in d21684dc4365cdacb2c697721190951967505ecb

File: src/dodal/devices/watsonmarlow323_pump.py
Comment (barnettwilliam) [reply src/dodal/devices/watsonmarlow323_pump.py]: Rebased and fixed in 84824ca3d5a6f0e30f8cc00fa40fec3cea06a2aa.

File: src/dodal/devices/watsonmarlow323_pump.py
Comment (DiamondJoseph) [inline src/dodal/devices/watsonmarlow323_pump.py:51]: nit: this could be done with the `add_children_as_readables` context, which is now the recommended way to do it [in the ophyd-async docs](https://blueskyproject.io/ophyd-async/main/how-to/make-a-simple-device.html#readable)

**Review by DiamondJoseph — APPROVED** [review]
nit, but otherwise looks good

**Comment by dperl-dls** [comment]
@barnettwilliam Do you still plan to merge this? Since you have addressed Callum's comments and Joseph has approved this you are free to dismiss Callum's review (or you could ask him to take a look again)

**Pull request merged**
